controllers/managedocs_controller.go

```diff
@@ -263,6 +270,23 @@ func (r *ManagedOCSReconciler) SetupWithManager(mgr ctrl.Manager, ctrlOptions *c
 		},
 	)
 
+	subPredicate := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(client client.Object) bool {
+				instance, ok := client.(*opv1a1.Subscription)
+				if !ok {
+					return false
+				}
+
+				// ignore if not a odf-operator subscription
+				if instance.Spec.Package != odfOLMPackageName {
+					return false
+				}
+				return true
```

Comment:
you can simplifyc and prevet ok form being created in func scope
```suggestion
				if instance, ok := client.(*opv1a1.Subscription); ok {
					// ignore if not a odf-operator subscription
					return  instance.Spec.Package != odfOLMPackageName
				} 
				return false
```